From db76829acd30d7cfaf03fb7ea76f1eaa420644ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20De=20Canni=C3=A8re?= <olivier.decanniere@qt.io>
Date: Wed, 22 Oct 2025 11:32:34 +0200
Subject: [PATCH 197/239] Compiler: Avoid generating duplicate settings names

The two computations may result in the same string.

Insert the second one only if it is differnt.

Amends c4455f1771483c8ed63d27454242f19b28b524b0

Change-Id: I0c4df0cf6bb5bba2c1cf6c50b8a7f105718b85ee
Reviewed-by: Sami Shalayel <sami.shalayel@qt.io>
(cherry picked from commit baaab2e4687e093c7fdd3e5049002b7d7e70ac18)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljsloggingutils.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/qmlcompiler/qqmljsloggingutils.cpp b/src/qmlcompiler/qqmljsloggingutils.cpp
index 4baf05c9e8..43712a976d 100644
--- a/src/qmlcompiler/qqmljsloggingutils.cpp
+++ b/src/qmlcompiler/qqmljsloggingutils.cpp
@@ -169,8 +169,10 @@ QString levelToString(const QQmlJS::LoggerCategory &category)
 static QStringList settingsNamesForCategory(const LoggerCategory &category)
 {
     const QString name = category.settingsName();
-    const QStringList result{ QStringLiteral("Warnings/") += name,
-                              QStringLiteral("Warnings/") += name.sliced(name.indexOf(u'.') + 1) };
+    QStringList result{ QStringLiteral("Warnings/") + name };
+    const QString sliced = "Warnings/"_L1 + name.sliced(name.indexOf(u'.') + 1);
+    if (sliced != result.last())
+        result.append(sliced);
     return result;
 }
 
-- 
2.51.2

